e2e_tests/tests/per_provider/normal_tests/export_key.rs

```diff
@@ -0,0 +1,212 @@
+zQmJfVfzYCVdr4X3Z/JOEexb3eu9p1Qj904sLu9Ds5NO7atT+qtDYVxgH5kQIrKk\
+mFNAx3NdAkEAovZ+DaorhkDiL/gFVzwoShyc1A6AWkH791sDlns2ETZ1WwE/ccYu\
+uJill/5XA9RKw6whUDzzNTsv7bFkCruAZwJARP5y6ALxz5DfFfbZuPU1d7/6g5Ki\
+b4fh8VzAV0ZbHa6hESLYBCbEdRE/WolvwfiGl0RBd6QxXTAYdPS46ODLLQJARrz4\
+urXDbuN7S5c9ukBCvOjuqp4g2Q0LcrPvOsMBFTeueXJxN9HvNfIM741X+DGOwqFV\
+VJ8gc1rd0y/NXVtGwQJAc2w23nTmZ/olcMVRia1+AFsELcCnD+JqaJ2AEF1Ng6Ix\
+V/X2l32v6t3B57sw/8ce3LCheEdqLHlSOpQiaD7Qfw==";
+
+#[test]
+fn export_key() -> Result<()> {
+    let mut client = TestClient::new();
+
+    if !client.is_operation_supported(Opcode::PsaExportKey) {
+        return Ok(());
+    }
+
+    let key_name = String::from("export_key");
+
+    client.generate_rsa_sign_key(key_name.clone())?;
+
+    let _ = client.export_key(key_name)?;
+
+    Ok(())
+}
+
+#[test]
+fn export_without_create() {
+    let mut client = TestClient::new();
+
+    if !client.is_operation_supported(Opcode::PsaExportKey) {
+        return;
+    }
+
+    let key_name = String::from("export_without_create");
+    let status = client
+        .export_key(key_name)
+        .expect_err("Key should not exist.");
+    assert_eq!(status, ResponseStatus::PsaErrorDoesNotExist);
+}
+
+#[test]
+fn import_and_export_public_key() -> Result<()> {
+    let mut client = TestClient::new();
+
+    if !client.is_operation_supported(Opcode::PsaExportKey) {
+        return Ok(());
+    }
+
+    let key_name = String::from("import_and_export_public_key");
+    let key_data = vec![
+        48, 129, 137, 2, 129, 129, 0, 153, 165, 220, 135, 89, 101, 254, 229, 28, 33, 138, 247, 20,
+        102, 253, 217, 247, 246, 142, 107, 51, 40, 179, 149, 45, 117, 254, 236, 161, 109, 16, 81,
+        135, 72, 112, 132, 150, 175, 128, 173, 182, 122, 227, 214, 196, 130, 54, 239, 93, 5, 203,
+        185, 233, 61, 159, 156, 7, 161, 87, 48, 234, 105, 161, 108, 215, 211, 150, 168, 156, 212,
+        6, 63, 81, 24, 101, 72, 160, 97, 243, 142, 86, 10, 160, 122, 8, 228, 178, 252, 35, 209,
+        222, 228, 16, 143, 99, 143, 146, 241, 186, 187, 22, 209, 86, 141, 24, 159, 12, 146, 44,
+        111, 254, 183, 54, 229, 109, 28, 39, 22, 141, 173, 85, 26, 58, 9, 128, 27, 57, 131, 2, 3,
+        1, 0, 1,
+    ];
+    client.import_rsa_public_key(key_name.clone(), key_data.clone())?;
+
+    assert_eq!(key_data, client.export_key(key_name)?);
+
+    Ok(())
+}
+
+#[test]
+fn check_rsa_export_format() -> Result<()> {
+    let mut client = TestClient::new();
+
+    if !client.is_operation_supported(Opcode::PsaExportKey) {
+        return Ok(());
+    }
+
+    let key_name = String::from("check_public_rsa_export_format");
+    client.generate_rsa_sign_key(key_name.clone())?;
+    let key = client.export_key(key_name)?;
+
+    // That should not fail if the bytes are in the expected format.
+    let _public_key: RSAPublicKey = picky_asn1_der::from_bytes(&key).unwrap();
+    let _private_key: RSAPrivateKey = picky_asn1_der::from_bytes(&key).unwrap();
```

Comment:
I don't think the implementation of [`RSAPrivateKey`](https://docs.rs/picky-asn1-x509/0.1.0/src/picky_asn1_x509/private_key_info.rs.html#141) is particularly safe - it doesn't fail on `from_bytes`, it will probably bomb out if you try to call `primes()` if the data you fed into it isn't valid. I've just tried it - if you feed in a public key and attempt to get the primes you get a panic.

We should probs raise a PR with them to change that to a proper implementation.